core/src/main/java/hudson/model/WorkspaceListener.java

```diff
@@ -3,25 +3,77 @@
 import hudson.ExtensionList;
 import hudson.ExtensionPoint;
 import hudson.FilePath;
+import hudson.Util;
+import hudson.console.ConsoleLogFilter;
+
+import java.io.OutputStream;
 
 public abstract class WorkspaceListener implements ExtensionPoint {
     
     /**
      * Called after a workspace is deleted successfully.
      * @param project
+     * @deprecated as of TODO_BEFORE_MERGE Use {@link #afterDelete(Job)}.
```

Comment:
> other comparable workflow refactoring to rely on Job/Run didn't used @Deprecated

Possible, I would have seen them I had put the same comment 😄 

> javadoc's @deprecated is well supported by all IDEs

Not sure about the javadoc, the annotation is supported for sure.

> I honestly never understood why this annotation was created 

I think this is a long discussion. For example, if at some point we want make the hpi-plugin to fail the build if there is use of deprecated methods, it will be easier to implement if we have annotations instead of javadocs.